packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx

```diff
@@ -0,0 +1,233 @@
+import { wordlists } from "ethers"
+import { FC, Suspense, useState } from "react"
+import { useNavigate } from "react-router-dom"
+import styled, { css, keyframes } from "styled-components"
+import useSWRImmutable from "swr/immutable"
+
+import { Button } from "../components/Button"
+import { IconBarWithIcons } from "../components/Recovery/IconBar"
+import { PageWrapper, Paragraph, Title } from "../components/Recovery/Page"
+import { FormError } from "../components/Typography"
+import { routes } from "../routes"
+import { useBackupRequired } from "../states/backupDownload"
+import { getSeedPhrase } from "../utils/messaging"
+
+const SeedPhraseGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  grid-template-rows: repeat(4, 1fr);
+  grid-gap: 12px;
+
+  margin-bottom: 96px;
+`
+
+const SeedWordBadge = styled.div`
+  padding: 4px 10px;
+  border-radius: 20px;
+  font-weight: 600;
+  font-size: 13px;
+  line-height: 18px;
+  text-align: center;
+  background-color: rgba(255, 255, 255, 0.1);
+`
+
+const pulseKeyframe = keyframes`
+  0% {
+    opacity: 1;
+  }
+  100% {
+    opacity: 0.2;
+  }
+`
+
+const LoadingSeedWordBadge = styled.div<{
+  animationDelay?: number
+}>`
+  height: 26px;
+  width: 100%;
+  border-radius: 20px;
+  background-color: rgba(255, 255, 255, 0.1);
+  animation: ${pulseKeyframe} 1s alternate infinite;
+  animation-delay: ${({ animationDelay = 0 }) => animationDelay}ms;
+`
+
+const SelectableSeedWordBadge = styled(SeedWordBadge)<{
+  isSelected: boolean
+}>`
+  cursor: pointer;
+  transition: all 0.2s ease-in-out;
+  user-select: none;
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.15);
+  }
+
+  ${({ isSelected }) =>
+    isSelected &&
+    css`
+      background-color: rgba(255, 255, 255, 0.4);
+      &:hover {
+        background-color: rgba(255, 255, 255, 0.4);
+      }
+    `}
+`
+
+function getRandomWord(excludeWordlist: string[]): string {
+  const excludeIndexes = excludeWordlist.map((word) =>
+    wordlists.en.getWordIndex(word),
+  )
```

Comment:
be more functional when you can, have you tried:

```suggestion
  const excludeIndexes = excludeWordlist.map(wordlists.en.getWordIndex)
```